### PR TITLE
Add support for Jupyter Lab v3.x and use Python 3 features for simplyfing syntax of the script to be executed remotely

### PR DESCRIPTION
**Pull request by airnandez**
With the introduction of Jupyter Lab v3, stackyter no longer works, because some commands were modified.

This pull request fixes that issue and uses features of Python 3 to improve the generation of the shell script to be executed at the remote host through the SSH tunnel. It also uses the recommended way to run a subprocess.

### CODE REVIEW

**Comment by johannct** [comment]
there are no issues related to this pull request. Is it possible to create one, or add here how to trigger the problem, so that we can exercise the solution?

**Comment by boutigny** [comment]
The simplest way to reproduce the problem is to run stackyter using a recent (> w_2021_04) release of the Rubin stack.

**Comment by airnandez** [comment]
You can observe what happens using this configuration file (`$HOME/.stackyter-config.yaml`):

```
default_config: 'cc'

cc:
  host:     'cca.in2p3.fr'
  jupyter:  'lab'
  runbefore:
    - 'source /cvmfs/sw.lsst.eu/linux-x86_64/lsst_distrib/w_2021_09/loadLSST.bash'
    - 'setup lsst_distrib'
```

**Comment by johannct** [comment]
ok what I observe then is an infinite loop with
`[NbserverListApp] WARNING | Config option `kernel_spec_manager_class` not recognized by `NbserverListApp`.
waiting...
`
is that the right symptom?

**Comment by airnandez** [comment]
The symptom is indeed that as a user **stackyter** cannot parse the output of Jupyter Lab v3, so it waits forever and does not give you the URL to use to access your notebook, even if the `jupyter lab` command is actually executed in the remote host.

The message 

```
[NbserverListApp] WARNING | Config option kernel_spec_manager_class not recognized by NbserverListApp.
```

is not related to **stackyter** is harmless albeit undesirable.

File: stackyter.py [hunk 1]
```diff
@@ -156,56 +158,77 @@ def get_config(config, configfile):
...
+
+    # Do we have to run something after sourcing the setup file ?
+    run_after = ''.join([run.replace("$", "\$") + "; " for run in args.runafter]) if args.runafter else ""
+
+    # Use the setup file given by the user to set up the working environment
+    user_setup = f"source {args.mysetup}" if args.mysetup else ""
+
+    script = f"""
+        #!/bin/bash
+        if [[ ! -d {args.workdir} ]]; then
```
[the remark this answers is not in the document]
Comment (johannct) [reply stackyter.py:175]: I guess something like that could work

[the remark this answers is not in the document]
Comment (johannct) [reply stackyter.py:175]: if [[ {args.workdir} = None ]];then
            cd $HOME
        elif [[ ! -d {args.workdir} ]]; then
            echo 'Error: directory {args.workdir} does not exist'
            exit 1
        else
            cd {args.workdir}
        fi

**Review by johannct — CHANGES_REQUESTED** [review]

**Comment by airnandez** [comment]
Setting the right default for the command line argument `workdir` works

**Review by johannct — APPROVED** [review]

**Pull request merged**